examples/expression-equality/expression_equality.md

```diff
@@ -0,0 +1,12 @@
+# Expression equality
+
+In Ballerina, expression equality is determined using two operators: `==` for value equality and `===` for reference equality. The `==` operator checks for deep equality between two values by comparing the actual data. In contrast, the `===` operator checks whether two values share the same storage identity, meaning it checks if the values reference the same memory location. 
+
+Values with storage identity, such as structured types like maps and arrays, are stored as references, so `===` can determine if two references point to the same underlying data. For simple types such as integers and booleans, which do not have storage identity, `===` behaves the same as `==`.
```

Comment:
Do we need to talk about behavioural (functions and objects) as well. They also has a storage identity

Reply:
Fallowon from `expression_equality.bal` I feel like we need to say about sequence values as well. Singleton once (ie. string and part of xml) don't have storage identities and others have (that is some `xml` have storage identity and others don't)

Reply:
Refer to https://github.com/ballerina-platform/ballerina-distribution/pull/5634/files#r1764870734

Reply:
I think for behavioural values, we can skip. For XML, we probably need a separate BBE, similar to the list equality BBE.